src/lib/format/l20n/ast/parser.js

```diff
@@ -101,16 +104,16 @@ class ParseContext {
     return entity;
   }
 
-  getValue(ch = this._source[this._index], optional = false) {
+  getValue(ch = this._source[this._index], index = false, required = true) {
```

Comment:
Do you actually need the third argument?  Could you reuse `hasIndex` for this check?  Also, perhaps keep the names of the args consistent with the names of variables in the caller code?